src/service/url-replacements-impl.js

```diff
@@ -453,6 +453,15 @@ export class GlobalVariableSource extends VariableSource {
       return win.navigator.userAgent;
     });
 
+    // Returns the user agent client hint.
+    this.setAsync(
+      'UACH',
+      (variable) =>
+        win.navigator?.userAgentData
+          ?.getHighEntropyValues([variable])
+          ?.then((values) => values[variable]) || Promise.resolve('')
```

Comment:
Are all of these values potentially null, or only `win.navigator`? If it's only the latter, you can write `win.navigator?.userAgentData.getHighEntropyValues(...).then(...)` etc, null chaining doesn't try to call any of the chained methods once it hits a null/undefined

Reply:
Either `win.navigator`, `win.navigator.userAgentData` or `win.navigator.userAgentData.getHighEntropyValues` can be null. The purpose is to filter out browsers that do not support the feature and return empty string.